src/node/createSetupServer.ts

```diff
@@ -150,10 +151,14 @@ export function createSetupServer(...interceptors: Interceptor[]) {
 
             // the node build will use the timers module to ensure @sinon/fake-timers or jest fake timers
             // don't affect this timeout.
-            setTimeout(() => {
-              resolve(mockedResponse)
+            const t = setTimeout(() => {
+              // Only resolve when the request is still relevant
+              if (pendingRequests[mockedRequest.id]) {
```

Comment:
Having this `if` statement was needed in node environments, but not in the service worker.
To keep things consistent (and just to be sure), I've also added it to the service worker.